third_party/terraform/resources/resource_compute_instance.go

```diff
@@ -395,6 +396,7 @@ func resourceComputeInstance() *schema.Resource {
 				Type:     schema.TypeMap,
 				Optional: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
```

Comment:
Out of curiousity, does it not normally default to schema.HashString?

Reply:
I'm pretty sure you're right. TBH I'm not sure where this got pulled in from.

Reply:
Also since labels isn't a Set, this doesn't actually do anything.